src/components/PoolingWidget/index.tsx

```diff
@@ -0,0 +1,192 @@
+          <React.Fragment key={stepName}>
+            <ProgressStep $bgColor={stepChecker(step, index) ? 'var(--color-background-progressBar)' : 'lightgrey'}>
+              <ProgressStepText $bold={stepChecker(step, index) ? 'bolder' : 'normal'}>{index + 1}</ProgressStepText>
+            </ProgressStep>
+            {index + 1 < 4 && (
+              <StepSeparator
+                $bgColor={stepChecker(step, index) ? 'var(--color-background-progressBar)' : 'lightgrey'}
+              />
+            )}
+          </React.Fragment>
+        ))}
+      </BarWrapper>
+      <BarWrapper $minHeight="auto">
+        {stepArray.map((stepName, index) => (
+          <React.Fragment key={stepName}>
+            <ProgressStepText $bold={stepChecker(step, index) ? 'bolder' : 'normal'}>{stepName}</ProgressStepText>
+            {index + 1 < 4 && <p />}
+          </React.Fragment>
+        ))}
+      </BarWrapper>
+    </>
+  )
+}
+
+const StepDescription: React.FC = () => (
+  <StepDescriptionWrapper>
+    <p>Setup your strategy once and allow your funds to be traded on your behalf.</p>
+    <ul>
+      <li>
+        <FontAwesomeIcon icon={faCheckCircle} />
+        No maintenance needed
+      </li>
+      <li>
+        <FontAwesomeIcon icon={faCheckCircle} />
+        No gas costs for trades
+      </li>
+      <li>
+        <FontAwesomeIcon icon={faCheckCircle} />
+        Cancellation possible at any time
+      </li>
+    </ul>
+    <a href="#">Learn more about strategies</a>
+  </StepDescriptionWrapper>
+)
+
+const StepTitle: React.FC<Pick<ProgressBarProps, 'step'>> = ({ step }) => {
+  const { title, subtext }: { title: string; subtext?: string } = useMemo(() => {
+    switch (step) {
+      case 1:
+        return {
+          title: '1. Select your trusted stablecoins',
+          subtext:
+            'Select two or more stablecoins you want to include in your liquidity provision and you believe are worth $1',
+        }
+      case 2:
+        return { title: '2. Define your spread', subtext: '' }
+      case 3:
+        return { title: '3. Create strategy', subtext: '' }
+      case 4:
+        return { title: '4. Add funding', subtext: '' }
+      default:
+        return { title: 'An error occurred, please try again' }
+    }
+  }, [step])
+
+  return (
+    <div>
+      <ProgressStepText as="h2" $bold="bolder">
+        {title}
+      </ProgressStepText>
+      {subtext && <GreySubText $justify="flex-start">{subtext}</GreySubText>}
+    </div>
+  )
+}
+
+function addRemoveItem(arr: number[], newItem: number): number[] {
+  if (!arr.includes(newItem)) return arr.concat(newItem)
+
+  return arr.reduce((acc: number[], item) => {
+    if (item === newItem) return acc
+
+    const newAcc = acc.concat(item)
+    return newAcc
+  }, [])
+}
+
+const PoolingInterface: React.FC = () => {
+  const [step, setStep] = useSafeState(1)
+  const [selectedTokens, setSelectedTokens] = useSafeState<number[]>([])
```

Comment:
What about using a `Set`? Don't know if you noticed, but I do like Sets :D

Reply:
Actually, a `Map<number, TokenDetails>` might be better, because on the next step (https://github.com/gnosis/dex-react/issues/402) we'll need to show token symbols, and I would prefer to keep the objects around instead of having to find them again by id.

Anyway, doesn't have to be a Map, just thinking of an easier interface to check what's selected for Step 1, while still keeping TokenDetails objects for Step 3.

What do you think?

Reply:
@alfetopito I had it setup this way before using a `Map<tokenID, tokenDetails>` but i had it coded in a way i guess that was super inefficient so the state updates were horrendously slow. like we're talking a few seconds to see the selected tokens become highlighted etc

I'll play around with it some more

Reply:
Dmitry helped me debug my "slow" update problem, he da best